### PR TITLE
ranges tutorial

### PR DESCRIPTION
**Pull request by h-2**
i will merge this very soonish, but please review in any case so that we can later fix stuff.

All examples are currently inline, I will move them out later.

### CODE REVIEW

File: doc/tutorial/ranges/index.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+# Motivation
+
+Traditionally most generic algorithms in the C++ standard library, like std::sort, take a pair of iterators.
+If you want to sort a std::vector `v`, you have to call `std::sort(v.begin(), v.end())` and not `std::sort(v)`.
+Why was this design with iterators chosen?
+It is more flexible, because it allows e.g.:
+  * sorting only all elements after the fifth one: `std::sort(v.begin() + 5, v.end())`
+  * using non-standard iterators like reverse iterators: `std::sort(v.rbegin() + 5, v.rend())` (sorts in reverse order)
+
+But this interface is less intuitive than just calling just calling std::sort on the entity that you wish to sort and
```
Comment (smehringer) [inline doc/tutorial/ranges/index.md]:
```suggestion
But this interface is less intuitive than just calling std::sort on the entity that you wish to sort and
```
:telephone_receiver:

File: doc/tutorial/ranges/index.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+We will discuss later what `std::view::reverse(v)` does, for now it is enough to understand that it returns something
+that appears like a container and that std::ranges::sort can sort it.
+Later we will see that this approach offers even more flexibility than working with iterators.
+
+# Ranges
+
+*Ranges* are an abstraction of "a collection of items", or "something iterable". The most basic definition
+requires only the existence of `begin()` and `end()` on the range.
+
+There are different ways to classify ranges, one way is through the capabilities of its iterator (the object
```
Comment (smehringer) [inline doc/tutorial/ranges/index.md]: I like that you explain an iterator but you have used iterator like 20 times beforehand without this explanation. So I would move this to the first mention of iterator or remove it.

File: doc/tutorial/ranges/index.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+**Containers** are the ranges most well known, they own their elements. SeqAn3 makes use of standard STL containers
+like `std::vector`, but also implements some custom containers.
+
+**Decorators** are ranges that are always defined on another range and decorate/annotate the underlying range
+with additional information. They do not own the underlying range, but can contain member data of their own.
+
+**Views** are ranges that are usually defined on another range and transform the underlying range
+via some algorithm or operation.
+Views do not own any data beyond their algorithm and the time it takes to destruct or copy them does not depend
```
[the remark this answers is not in the document]
Comment (h-2) [reply doc/tutorial/ranges/index.md]: it's not required for construction actually :roll_eyes:

[the remark this answers is not in the document]
Comment (h-2) [reply doc/tutorial/ranges/index.md]: but i'll change it anyway,...

**Pull request merged**